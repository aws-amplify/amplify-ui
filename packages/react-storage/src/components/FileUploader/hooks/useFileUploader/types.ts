import { FileStatus, StorageFiles } from '../../types';
import { UploadTask } from '../../utils';

export interface UseFileUploaderState {
  files: StorageFiles;
}

export enum FileUploaderActionTypes {
  ADD_FILES = 'ADD_FILES',
  CLEAR_FILES = 'CLEAR_FILES',
  QUEUE_FILES = 'QUEUE_FILES',
  SET_STATUS = 'SET_STATUS',
  SET_PROCESSED_FILE_KEY = 'SET_PROCESSED_FILE_KEY',
  SET_STATUS_UPLOADING = 'SET_STATUS_UPLOADING',
  SET_UPLOAD_PROGRESS = 'SET_UPLOAD_PROGRESS',
  REMOVE_UPLOAD = 'REMOVE_UPLOAD',
}

export type GetFileErrorMessage = (file: File) => string;

export type Action =
  | {
      type: FileUploaderActionTypes.ADD_FILES;
      files: File[];
      status: FileStatus;
      getFileErrorMessage: GetFileErrorMessage;
    }
  | {
      type: FileUploaderActionTypes.CLEAR_FILES;
    }
  | {
      type: FileUploaderActionTypes.SET_STATUS;
      id: string;
      status: FileStatus;
    }
  | {
      type: FileUploaderActionTypes.QUEUE_FILES;
    }
  | {
      type: FileUploaderActionTypes.SET_STATUS_UPLOADING;
      id: string;
      uploadTask?: UploadTask;
    }
  | {
      type: FileUploaderActionTypes.SET_UPLOAD_PROGRESS;
      id: string;
      progress: number;
    }
  | {
      type: FileUploaderActionTypes.SET_PROCESSED_FILE_KEY;
      id: string;
      processedKey: string;
    }
  | {
      type: FileUploaderActionTypes.REMOVE_UPLOAD;
      id: string;
    };

export interface AddFilesActionParams {
  files: File[];
  status: FileStatus;
  getFileErrorMessage: GetFileErrorMessage;
}
