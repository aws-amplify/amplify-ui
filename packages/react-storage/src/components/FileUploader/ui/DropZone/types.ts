import { FileUploaderDisplayText } from '../../utils/displayText';

export interface DropZoneProps {
  children?: React.ReactNode;
  displayText: FileUploaderDisplayText;
  inDropZone: boolean;
  onDragEnter: (event: React.DragEvent<HTMLDivElement>) => void;
  onDragLeave: (event: React.DragEvent<HTMLDivElement>) => void;
  onDragOver: (event: React.DragEvent<HTMLDivElement>) => void;
  onDragStart: (event: React.DragEvent<HTMLDivElement>) => void;
  onDrop: (event: React.DragEvent<HTMLDivElement>) => void;
  testId?: string;
}
