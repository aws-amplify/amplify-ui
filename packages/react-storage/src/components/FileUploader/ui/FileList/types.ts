import { FileUploaderDisplayTextDefault, TaskHandler } from '../../utils';
import { FileStatus, StorageFile } from '../../types';

export interface FileListProps {
  displayText: FileUploaderDisplayTextDefault;
  files: StorageFile[];
  isResumable: boolean;
  onCancelUpload: TaskHandler;
  onDeleteUpload: (params: { id: string }) => void;
  onPause: TaskHandler;
  onResume: TaskHandler;
  showThumbnails: boolean;
  hasMaxFilesError: boolean;
  maxFileCount: number;
}

export interface FileControlProps {
  displayText: FileUploaderDisplayTextDefault;
  displayName: string;
  errorMessage: string;
  isImage: boolean;
  isResumable: boolean;
  isUploading: boolean;
  loaderIsDeterminate: boolean;
  onRemove: () => void;
  onPause: () => void;
  onResume: () => void;
  progress: number;
  showThumbnails: boolean;
  size?: number;
  status: FileStatus;
  thumbnailUrl: string;
}

export interface FileStatusMessageProps
  extends Pick<
    FileUploaderDisplayTextDefault,
    'getUploadingText' | 'getPausedText' | 'uploadSuccessfulText'
  > {
  status: FileStatus;
  errorMessage: string;
  percentage: number;
}

export interface UploadDetailsProps {
  displayName: string;
  fileSize?: number;
}

export interface FileThumbnailProps {
  isImage: boolean;
  fileName: string;
  url: string;
}

export interface FileRemoveButtonProps {
  altText: string;
  onClick: () => void;
}
