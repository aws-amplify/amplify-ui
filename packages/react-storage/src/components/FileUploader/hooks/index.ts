export { useFileUploader, UseFileUploader } from './useFileUploader';
export { useUploadFiles } from './useUploadFiles';
