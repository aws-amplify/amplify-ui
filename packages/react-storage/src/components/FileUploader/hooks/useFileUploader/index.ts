export { useFileUploader, UseFileUploader } from './useFileUploader';
