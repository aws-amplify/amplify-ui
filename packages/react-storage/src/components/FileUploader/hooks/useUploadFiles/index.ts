export type { UseUploadFilesProps } from './useUploadFiles';
export { useUploadFiles } from './useUploadFiles';
