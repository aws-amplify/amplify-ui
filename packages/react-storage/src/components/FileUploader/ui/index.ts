export { Container, ContainerProps } from './Container';
export { DropZone, DropZoneProps } from './DropZone';
export { FileList, FileListProps } from './FileList';
export { FileListHeader, FileListHeaderProps } from './FileListHeader';
export { FileListFooter, FileListFooterProps } from './FileListFooter';
export { FilePicker, FilePickerProps } from './FilePicker';
