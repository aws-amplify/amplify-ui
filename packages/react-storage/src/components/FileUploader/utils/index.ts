export { checkMaxFileSize } from './checkMaxFileSize';
export {
  defaultFileUploaderDisplayText,
  FileUploaderDisplayText,
  FileUploaderDisplayTextDefault,
} from './displayText';
export { filterAllowedFiles } from './filterAllowedFiles';
export { getInput } from './getInput';

export {
  PathCallback,
  TaskEvent,
  TaskHandler,
  uploadFile,
  UploadTask,
} from './uploadFile';
