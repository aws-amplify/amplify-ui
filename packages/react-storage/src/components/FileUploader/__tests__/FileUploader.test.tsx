import React from 'react';
import {
  fireEvent,
  render,
  waitFor,
  act,
  getByTestId,
} from '@testing-library/react';
import * as Storage from 'aws-amplify/storage';

import { ComponentClassName } from '@aws-amplify/ui';

import * as StorageHooks from '../hooks';
import {
  FileUploader,
  MISSING_REQUIRED_PROPS_MESSAGE,
  ACCESS_LEVEL_DEPRECATION_MESSAGE,
} from '../FileUploader';
import { FileUploaderProps, FileUploaderHandle, FileStatus } from '../types';
import { defaultFileUploaderDisplayText } from '../utils';

const warnSpy = jest.spyOn(console, 'warn').mockImplementation();

const uploadDataSpy = jest
  .spyOn(Storage, 'uploadData')
  .mockImplementation((input) => ({
    cancel: jest.fn(),
    pause: jest.fn(),
    resume: jest.fn(),
    state: 'SUCCESS',
    result: Promise.resolve({ key: input.key, data: input.data }),
  }));

const fileUploaderProps: FileUploaderProps = {
  accessLevel: 'guest',
  maxFileCount: 100,
};

describe('FileUploader', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('behaves as expected with an accessLevel prop', () => {
    const { container, getByText } = render(
      <FileUploader {...fileUploaderProps} />
    );
    expect(container).toMatchSnapshot();

    expect(
      container.getElementsByClassName(
        `${ComponentClassName.FileUploaderDropZone}`
      )
    ).toHaveLength(1);

    expect(
      container.getElementsByClassName(
        `${ComponentClassName.FileUploaderDropZoneText}`
      )
    ).toHaveLength(1);

    expect(
      container.getElementsByClassName(
        `${ComponentClassName.FileUploaderDropZoneIcon}`
      )
    ).toHaveLength(1);

    expect(
      container.getElementsByClassName(
        `${ComponentClassName.FileUploaderFilePicker}`
      )
    ).toHaveLength(1);

    expect(
      getByText(defaultFileUploaderDisplayText.browseFilesText)
    ).toBeVisible();
    expect(
      getByText(defaultFileUploaderDisplayText.dropFilesText)
    ).toBeVisible();

    expect(warnSpy).toHaveBeenCalledTimes(1);
  });

  it('behaves as expected with a path prop', () => {
    const { container, getByText } = render(
      <FileUploader maxFileCount={3} path={() => 'my-path'} />
    );
    expect(container).toMatchSnapshot();

    expect(
      container.getElementsByClassName(
        `${ComponentClassName.FileUploaderDropZone}`
      )
    ).toHaveLength(1);

    expect(
      container.getElementsByClassName(
        `${ComponentClassName.FileUploaderDropZoneText}`
      )
    ).toHaveLength(1);

    expect(
      container.getElementsByClassName(
        `${ComponentClassName.FileUploaderDropZoneIcon}`
      )
    ).toHaveLength(1);

    expect(
      container.getElementsByClassName(
        `${ComponentClassName.FileUploaderFilePicker}`
      )
    ).toHaveLength(1);

    expect(
      getByText(defaultFileUploaderDisplayText.browseFilesText)
    ).toBeVisible();
    expect(
      getByText(defaultFileUploaderDisplayText.dropFilesText)
    ).toBeVisible();

    expect(warnSpy).not.toHaveBeenCalled();
  });

  it('renders as expected with autoUpload turned off', () => {
    const { getByText } = render(
      <FileUploader {...fileUploaderProps} autoUpload={false} />
    );
    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    const mockFile = new File(['hello'], 'hello.png', { type: 'image/png' });
    fireEvent.change(hiddenInput, { target: { files: [mockFile] } });

    expect(
      getByText(defaultFileUploaderDisplayText.clearAllButtonText)
    ).toBeVisible();
    expect(
      getByText(defaultFileUploaderDisplayText.getSelectedFilesText(1))
    ).toBeVisible();
  });

  it('renders as expected with override display text', () => {
    const displayText = {
      ...defaultFileUploaderDisplayText,
      dropFilesText: 'Drag and drop files here, or click to select files',
      browseFilesText: 'Select Files',
    };
    const { getByText } = render(
      <FileUploader {...fileUploaderProps} displayText={displayText} />
    );
    expect(
      getByText('Drag and drop files here, or click to select files')
    ).toBeVisible();
    expect(getByText('Select Files')).toBeVisible();
  });

  it('displays error message when file exceeds max file size', () => {
    const maxFileSize = 0;
    const { getByText } = render(
      <FileUploader {...fileUploaderProps} maxFileSize={maxFileSize} />
    );
    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();

    const mockFile = new File(['hello'], 'hello.png', { type: 'image/png' });
    fireEvent.change(hiddenInput, { target: { files: [mockFile] } });

    expect(
      getByText(defaultFileUploaderDisplayText.getFileSizeErrorText('0 B'))
    ).toBeVisible();
  });

  it('displays error message when max file count is exceeded', () => {
    const maxFileCount = 1;
    const { getByText } = render(
      <FileUploader {...fileUploaderProps} maxFileCount={maxFileCount} />
    );
    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();

    const mockFile = new File(['hello'], 'hello.png', { type: 'image/png' });
    const mockFile2 = new File(['hello2'], 'hello2.png', { type: 'image/png' });
    fireEvent.change(hiddenInput, { target: { files: [mockFile, mockFile2] } });

    expect(
      getByText(
        defaultFileUploaderDisplayText.getMaxFilesErrorText(maxFileCount)
      )
    ).toBeVisible();
  });

  it('calls onUploadSuccess callback when file is successfully uploaded', async () => {
    const onUploadSuccess = jest.fn();
    render(
      <FileUploader {...fileUploaderProps} onUploadSuccess={onUploadSuccess} />
    );
    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();
    const file = new File(['file content'], 'file.txt', { type: 'text/plain' });
    fireEvent.change(hiddenInput, {
      target: { files: [file] },
    });

    // Wait for the file to be uploaded
    await waitFor(() => {
      expect(uploadDataSpy).toHaveBeenCalledWith({
        key: file.name,
        data: file,
        options: {
          accessLevel: 'guest',
          contentType: 'text/plain',
          onProgress: expect.any(Function),
        },
      });
      expect(onUploadSuccess).toHaveBeenCalledTimes(1);
    });
  });

  it('calls onUploadStart callback when file starts uploading', async () => {
    const onUploadStart = jest.fn();
    render(
      <FileUploader {...fileUploaderProps} onUploadStart={onUploadStart} />
    );
    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();
    const file = new File(['file content'], 'file.txt', { type: 'text/plain' });
    fireEvent.change(hiddenInput, {
      target: { files: [file] },
    });

    // Wait for the file to be uploaded
    await waitFor(() => {
      expect(uploadDataSpy).toHaveBeenCalledWith({
        key: file.name,
        data: file,
        options: {
          accessLevel: 'guest',
          contentType: 'text/plain',
          onProgress: expect.any(Function),
        },
      });
      expect(onUploadStart).toHaveBeenCalledTimes(1);
    });
  });

  it('provides the correct file key on a remove file event before upload', () => {
    const onFileRemove = jest.fn();

    const { container } = render(
      <FileUploader
        {...fileUploaderProps}
        autoUpload={false}
        onFileRemove={onFileRemove}
      />
    );

    const hiddenInput: HTMLInputElement = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();

    const file = new File(['file content'], 'file.txt', { type: 'text/plain' });

    fireEvent.change(hiddenInput, { target: { files: [file] } });

    expect(uploadDataSpy).not.toHaveBeenCalled();

    const removeButton = getByTestId(
      container,
      'storage-manager-remove-button'
    );
    expect(removeButton).toBeDefined();

    fireEvent.click(removeButton);

    expect(onFileRemove).toHaveBeenCalledTimes(1);
    expect(onFileRemove).toHaveBeenCalledWith({ key: file.name });
  });

  it('provides the correct file key on a remove file event after upload', async () => {
    const onFileRemove = jest.fn();

    const { container } = render(
      <FileUploader {...fileUploaderProps} onFileRemove={onFileRemove} />
    );
    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();
    const file = new File(['file content'], 'file.txt', { type: 'text/plain' });
    fireEvent.change(hiddenInput, {
      target: { files: [file] },
    });

    // Wait for the file to be uploaded
    await waitFor(() => {
      expect(uploadDataSpy).toHaveBeenCalled();

      const removeButton = getByTestId(
        container,
        'storage-manager-remove-button'
      );
      expect(removeButton).toBeDefined();

      fireEvent.click(removeButton);

      expect(onFileRemove).toHaveBeenCalledTimes(1);
      expect(onFileRemove).toHaveBeenCalledWith({ key: file.name });
    });
  });

  it('provides the processed file key on a remove file event after upload when processFile is provided', async () => {
    const onFileRemove = jest.fn();

    const processedKey = 'processedKey';
    const processFile: FileUploaderProps['processFile'] = (input) => ({
      ...input,
      key: processedKey,
    });

    const { container } = render(
      <FileUploader
        {...fileUploaderProps}
        onFileRemove={onFileRemove}
        processFile={processFile}
      />
    );

    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();
    const file = new File(['file content'], 'file.txt', { type: 'text/plain' });

    fireEvent.change(hiddenInput, { target: { files: [file] } });

    // Wait for the file to be uploaded
    await waitFor(() => {
      expect(uploadDataSpy).toHaveBeenCalled();

      const removeButton = getByTestId(
        container,
        'storage-manager-remove-button'
      );
      expect(removeButton).toBeDefined();

      fireEvent.click(removeButton);

      expect(onFileRemove).toHaveBeenCalledTimes(1);
      expect(onFileRemove).toHaveBeenCalledWith({ key: processedKey });
    });
  });

  it('provides the processed file key on a remove file event after upload when processFile is provided with a path function', async () => {
    const onFileRemove = jest.fn();

    const processedKey = 'processedKey';
    const processFile: FileUploaderProps['processFile'] = (input) => ({
      ...input,
      key: processedKey,
    });

    const { container } = render(
      <FileUploader
        {...fileUploaderProps}
        onFileRemove={onFileRemove}
        processFile={processFile}
        path={() => 'my-path'}
        accessLevel={undefined}
      />
    );

    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();
    const file = new File(['file content'], 'file.txt', { type: 'text/plain' });

    fireEvent.change(hiddenInput, { target: { files: [file] } });

    // Wait for the file to be uploaded
    await waitFor(() => {
      expect(uploadDataSpy).toHaveBeenCalled();

      const removeButton = getByTestId(
        container,
        'storage-manager-remove-button'
      );
      expect(removeButton).toBeDefined();

      fireEvent.click(removeButton);

      expect(onFileRemove).toHaveBeenCalledTimes(1);
      expect(onFileRemove).toHaveBeenCalledWith({ key: processedKey });
    });
  });

  it('logs a warning if maxFileCount is zero', () => {
    render(<FileUploader {...fileUploaderProps} maxFileCount={0} />);

    expect(warnSpy).toHaveBeenCalledTimes(2);
    expect(warnSpy.mock.calls[0][0]).toBe(MISSING_REQUIRED_PROPS_MESSAGE);
    expect(warnSpy.mock.calls[1][0]).toBe(ACCESS_LEVEL_DEPRECATION_MESSAGE);
  });

  it('logs a warning if provided an accessLevel prop', () => {
    render(<FileUploader {...fileUploaderProps} maxFileCount={1} />);

    expect(warnSpy).toHaveBeenCalledTimes(1);
    expect(warnSpy.mock.calls[0][0]).toBe(ACCESS_LEVEL_DEPRECATION_MESSAGE);
  });

  it('should trigger hidden input onChange', async () => {
    const mockAddFiles = jest.fn();
    jest.spyOn(StorageHooks, 'useFileUploader').mockReturnValue({
      addFiles: mockAddFiles,
      files: [],
      status: FileStatus.QUEUED,
    } as unknown as StorageHooks.UseFileUploader);

    const { findByRole } = render(<FileUploader {...fileUploaderProps} />);

    const filePickerButton = await findByRole('button');

    expect(filePickerButton).toBeInTheDocument();

    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();

    fireEvent.click(filePickerButton);

    expect(hiddenInput).toHaveValue('');

    const mockFile = new File(['hello'], 'hello.png', { type: 'image/png' });

    fireEvent.change(hiddenInput, { target: { files: [mockFile] } });

    expect(mockAddFiles).toHaveBeenCalledTimes(1);
    expect(mockAddFiles).toHaveBeenCalledWith({
      files: [mockFile],
      status: FileStatus.QUEUED,
      getFileErrorMessage: expect.any(Function),
    });
  });

  it('clears files when imperative handle clearFiles() is called', () => {
    const ref = React.createRef<FileUploaderHandle>();
    const mockAddFiles = jest.fn();
    const mockClearFiles = jest.fn();
    jest.spyOn(StorageHooks, 'useFileUploader').mockReturnValue({
      addFiles: mockAddFiles,
      clearFiles: mockClearFiles,
      files: [],
    } as unknown as StorageHooks.UseFileUploader);

    render(<FileUploader {...fileUploaderProps} ref={ref} />);
    const hiddenInput = document.querySelector(
      'input[type="file"]'
    ) as HTMLInputElement;

    expect(hiddenInput).toBeInTheDocument();
    const file = new File(['file content'], 'file.txt', { type: 'text/plain' });
    fireEvent.change(hiddenInput, {
      target: { files: [file] },
    });

    expect(mockAddFiles).toHaveBeenCalledTimes(1);
    expect(mockAddFiles).toHaveBeenCalledWith({
      files: [file],
      status: FileStatus.QUEUED,
      getFileErrorMessage: expect.any(Function),
    });

    act(() => ref.current?.clearFiles());
    expect(mockClearFiles).toHaveBeenCalledTimes(1);
    expect(mockClearFiles).toHaveBeenCalledWith();
  });
});
