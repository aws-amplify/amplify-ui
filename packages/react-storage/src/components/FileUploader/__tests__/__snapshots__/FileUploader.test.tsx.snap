// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploader behaves as expected with a path prop 1`] = `
<div>
  <div
    class="amplify-fileuploader "
  >
    <div
      class="amplify-fileuploader__dropzone"
    >
      <span
        aria-hidden="true"
        class="amplify-fileuploader__dropzone__icon"
      >
        <span
          class="amplify-icon"
          style="width: 1em; height: 1em;"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11zM8 15.01l1.41 1.41L11 14.84V19h2v-4.16l1.59 1.59L16 15.01 12.01 11 8 15.01z"
              fill="currentColor"
            />
          </svg>
        </span>
      </span>
      <p
        class="amplify-text amplify-fileuploader__dropzone__text"
      >
        Drop files here or
      </p>
      <button
        class="amplify-button amplify-field-group__control amplify-button--small amplify-fileuploader__file__picker"
        type="button"
      >
        Browse files
      </button>
      <span
        class="amplify-visually-hidden"
      >
        <input
          accept=""
          multiple=""
          tabindex="-1"
          type="file"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`FileUploader behaves as expected with an accessLevel prop 1`] = `
<div>
  <div
    class="amplify-fileuploader "
  >
    <div
      class="amplify-fileuploader__dropzone"
    >
      <span
        aria-hidden="true"
        class="amplify-fileuploader__dropzone__icon"
      >
        <span
          class="amplify-icon"
          style="width: 1em; height: 1em;"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11zM8 15.01l1.41 1.41L11 14.84V19h2v-4.16l1.59 1.59L16 15.01 12.01 11 8 15.01z"
              fill="currentColor"
            />
          </svg>
        </span>
      </span>
      <p
        class="amplify-text amplify-fileuploader__dropzone__text"
      >
        Drop files here or
      </p>
      <button
        class="amplify-button amplify-field-group__control amplify-button--small amplify-fileuploader__file__picker"
        type="button"
      >
        Browse files
      </button>
      <span
        class="amplify-visually-hidden"
      >
        <input
          accept=""
          multiple=""
          tabindex="-1"
          type="file"
        />
      </span>
    </div>
  </div>
</div>
`;
